sw/device/tests/dif/dif_pinmux_test.cc

```diff
@@ -0,0 +1,379 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "gtest/gtest.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/testing/mock_mmio.h"
+
+extern "C" {
+#include "pinmux_regs.h"  // Generated.
+#include "sw/device/lib/dif/dif_pinmux.h"
+}
+
+namespace dif_pinmux_test {
+namespace {
+using testing::Test;
+using mock_mmio::MmioTest;
+
+class DifPinmuxTest : public Test, public MmioTest {
+ protected:
+  void DeviceInit() {
+    EXPECT_READ32(PINMUX_REGEN_REG_OFFSET, 1);
+
+    // Reset sequence.
+    std::vector<ptrdiff_t> insel_registers{
+        PINMUX_PERIPH_INSEL0_REG_OFFSET, PINMUX_PERIPH_INSEL1_REG_OFFSET,
+        PINMUX_PERIPH_INSEL2_REG_OFFSET, PINMUX_PERIPH_INSEL3_REG_OFFSET,
+        PINMUX_PERIPH_INSEL4_REG_OFFSET, PINMUX_PERIPH_INSEL5_REG_OFFSET,
+        PINMUX_PERIPH_INSEL6_REG_OFFSET,
+    };
+    for (const auto &reg : insel_registers) {
+      EXPECT_WRITE32(reg, 0);
+    }
+
+    std::vector<ptrdiff_t> outsel_registers{
+        PINMUX_MIO_OUTSEL0_REG_OFFSET, PINMUX_MIO_OUTSEL1_REG_OFFSET,
+        PINMUX_MIO_OUTSEL2_REG_OFFSET, PINMUX_MIO_OUTSEL3_REG_OFFSET,
+        PINMUX_MIO_OUTSEL4_REG_OFFSET, PINMUX_MIO_OUTSEL5_REG_OFFSET,
+        PINMUX_MIO_OUTSEL6_REG_OFFSET,
+    };
+    for (const auto &reg : outsel_registers) {
+      // TODO - autogenerate?
```

Comment:
Is this autogenerateable? I'm not sure given what our hjsons currently carry. I'd drop the comment.

Reply:
Done.